handlers/attachment_resource.go

```diff
@@ -160,6 +160,87 @@ func HandleGetAttachmentDetails(svc dao.Service) http.Handler {
 	})
 }
 
+// HandleDownloadAttachment download an attachment which is stored against an Insolvency case
+func HandleDownloadAttachment(svc dao.Service) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		vars := mux.Vars(req)
+		transactionID := utils.GetTransactionIDFromVars(vars)
+		if transactionID == "" {
+			log.ErrorR(req, fmt.Errorf("there is no transaction ID in the URL path"))
+			m := models.NewMessageResponse("transaction ID is not in the URL path")
+			utils.WriteJSONWithStatus(w, req, m, http.StatusBadRequest)
+			return
+		}
+
+		attachmentID := utils.GetAttachmentIDFromVars(vars)
+		if attachmentID == "" {
+			log.ErrorR(req, fmt.Errorf("there is no attachment ID in the URL path"))
+			m := models.NewMessageResponse("attachment ID is not in the URL path")
+			utils.WriteJSONWithStatus(w, req, m, http.StatusBadRequest)
+			return
+		}
+
+		// Get attachment data from DB to check if attachment ID is valid
+		attachmentResource, err := svc.GetAttachmentFromInsolvencyResource(transactionID, attachmentID)
+		if err != nil {
+			log.ErrorR(req, fmt.Errorf("failed to get attachment from insolvency db resource for transaction [%s] with attachment id [%s]: %v", transactionID, attachmentID, err))
+			m := models.NewMessageResponse("there was a problem handling your request")
+			utils.WriteJSONWithStatus(w, req, m, http.StatusInternalServerError)
+			return
+		}
+		if attachmentResource == (models.AttachmentResourceDao{}) {
+			m := models.NewMessageResponse("attachment id is not valid")
+			utils.WriteJSONWithStatus(w, req, m, http.StatusBadRequest)
```

Comment:
We should be telling the user that the `fmt.Sprintf("attachment with id [%s] is not found", attachmentID)`
And we should be returning a 404

```suggestion
			utils.WriteJSONWithStatus(w, req, m, http.StatusNotFound)
```